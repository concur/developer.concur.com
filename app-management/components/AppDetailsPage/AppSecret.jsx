import React, { PropTypes } from 'react';

const AppSecret = ({ clickHandler, secret }) => (
  <div className="alert alert-warning">
    <p>
      You can regenerate your clientSecret below.
      This process will clear your old clientSecret and require updated to your
      apps or scripts.
    </p>
    <button
      type="button"
      className="btn orange"
      onClick={clickHandler}
    >
      Regenerate Application clientSecret
    </button>
    {secret &&
      <p>
        <strong>
          Here is your new clientSecret.
        </strong>
        <br />
        <code>{secret}</code>
      </p>
    }
  </div>
);

AppSecret.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  secret: PropTypes.string,
};

export default AppSecret;
