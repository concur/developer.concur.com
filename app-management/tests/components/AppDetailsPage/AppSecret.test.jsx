import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AppSecret from '../../../components/AppDetailsPage/AppSecret';

describe('<AppSecret />', () => {
  const defaultProps = {
    clickHandler: jest.fn(),
    secret: 'a-test-secret',
  };

  it('should call the click handler when "Get New Secret" is pressed', () => {
    const secret = shallow(<AppSecret {...defaultProps} />);
    secret.find('button').simulate('click');

    expect(defaultProps.clickHandler).toHaveBeenCalled();
    expect(shallowToJson(secret)).toMatchSnapshot();
  });

  it('should not display the secret if not provided', () => {
    const props = {
      ...defaultProps,
      secret: null,
    };
    const secret = shallow(<AppSecret {...props} />);

    expect(shallowToJson(secret)).toMatchSnapshot();
  });
});
