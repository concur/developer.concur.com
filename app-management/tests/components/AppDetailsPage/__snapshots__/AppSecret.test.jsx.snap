exports[`<AppSecret /> should call the click handler when "Get New Secret" is pressed 1`] = `
<div
  className="alert alert-warning">
  <p>
    You can regenerate your clientSecret below. This process will clear your old clientSecret and require updated to your apps or scripts.
  </p>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button">
    Regenerate Application clientSecret
  </button>
  <p>
    <strong>
      Here is your new clientSecret.
    </strong>
    <br />
    <code>
      a-test-secret
    </code>
  </p>
</div>
`;

exports[`<AppSecret /> should not display the secret if not provided 1`] = `
<div
  className="alert alert-warning">
  <p>
    You can regenerate your clientSecret below. This process will clear your old clientSecret and require updated to your apps or scripts.
  </p>
  <button
    className="btn orange"
    onClick={[Function]}
    type="button">
    Regenerate Application clientSecret
  </button>
</div>
`;
